twilight-standby/src/lib.rs

```diff
@@ -165,44 +154,70 @@ impl Standby {
     /// This function must be called when events are received in order for
     /// futures returned by methods to fulfill.
     pub fn process(&self, event: &Event) -> ProcessResults {
-        tracing::trace!(event_type = ?event.kind(), ?event, "processing event");
-
         let mut completions = ProcessResults::new();
 
         match event {
             Event::InteractionCreate(e) => {
                 if e.kind == InteractionType::MessageComponent
                     && let Some(message) = &e.message
                 {
-                    completions.add_with(&Self::process_specific_event(
-                        &self.components,
-                        message.id,
-                        e,
-                    ));
+                    let remove =
+                        self.components
+                            .get_mut(&message.id)
+                            .is_some_and(|mut bystanders| {
+                                completions.add_with(&Self::bystander_iter(&mut bystanders, e));
+                                bystanders.is_empty()
+                            });
+                    if remove {
+                        self.components.remove(&message.id);
+                    }
```

Comment:
I think these 4 blocks should probably be sent to a function like:
```rust
    fn process_event<IdKind, E: Clone>(
        map: &DashMap<Id<IdKind>, Vec<Bystander<E>>>,
        id: Id<IdKind>,
        completions: &mut ProcessResults,
        e: &E,
    ) {
        let remove = map.get_mut(&id).is_some_and(|mut bystanders| {
            completions.add_with(&Self::bystander_iter(&mut bystanders, e));
            bystanders.is_empty()
        });
        if remove {
            map.remove(&id);
        }
    }
```
instead, no reason to duplicate the same code 4 times.